### PR TITLE
add "init templates" cmd, used to list templates in format

### LINKED ISSUE #265: [Feature Request] Kusion provides API for integration tools
## Feature Request

**Is your feature request related to a problem? Please describe:**

Now the extension is accomplishing its goals by executing the kusion binary for each time and then resolve the `stdout & stderr`, but since the stdout/stderr of kusion binary are not designed for tools integration, the output messages like this are disturbing:
image

the vscode-kusion extension requires kusion to provide APIs for tools to integrate kusion's capacities more elegantly and normatively. 

**Describe the feature you'd like:**

At least a few Go APIs of kusion are required:
- compile
- preview
- apply
- watch
- list projects
- list stacks
- list templates

### ListTemplates API

- API id：`init.ListTemplates`
- params:

| name | type | required | default value | description | example |
| ----- | ----- | -------- | ------------- | ----------- | --------- |
| online | boolean | optional | false | determines if kusion downloads templates from online repo | `true`, `false` |
| url | string | optional | `"https://github.com/KusionStack/kusion-templates"` | the online repo url, meaningful only when `online==true`|  |
| fsPath | string | optional |  | the fsPath to find the templates, meaningful only when `online == false` |`/path/to/templates` |

- returns:

| name | type | description | example |
| ----- | ----- | ----------- | -------- |
| templateList | Template[] | list of templates found | |
| err | Error | error code and message | `{"code":"invalidPath"， "message":"path not exists (or repo url not exists)"}` |

```go
type Template struct {
    name string
    description string
    content string // now the template could be the content of `kusion.yaml` file under the template directory: for example https://github.com/KusionStack/kusion-templates/blob/main/code-city/kusion.yaml
}

type Error struct {
    code  string
    message string
}
```

### PR DESCRIPTION
**Pull request by healthjyk**
#### What type of PR is this?
feature

#### What this PR does / why we need it:
list init templates from local or online, support json format. Hence, user can parse the specific elements of the templates.

#### Which issue(s) this PR fixes:

Fixes #265 

#### Special notes for your reviewer:
If the cmd failed, the output is not in json format, even if "--output=json" is specified.

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., design docs, usage docs, etc.:

```docs
kusion init templates command helps you get the templates which are used to generate a scaffolding KCL project.
Here are the examples:
# Get name and description of internal templates
  kusion init templates
# Get templates from specific templates location, and output in json format
  kusion init templates https://github.com/<user>/<repo> --online=true --output=json
```

### CODE REVIEW

**Comment by healthjyk** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by SparkYuan** [comment]
@healthjyk please add an example output for this command

**Comment by healthjyk** [comment]
an successful output is like:

**Comment by healthjyk** [comment]
> @healthjyk please add an example output for this command

an successful request is like:

**Comment by healthjyk** [comment]
> @healthjyk please add an example output for this command

an successful request is like:
[
    {
        "dir": "/${userName}/.kusionup/current/templates/external/code-city",
        "name": "code-city",
        "projectName": "code-city",
        "description": "Code City metaphor for visualizing Go source code in 3D.",
        "quickstart": "kusion compile main.k -Y ci-test/settings.yaml",
        "projectFields": [
            {
                "name": "AppName",
                "description": "App name. Default to 'gocity'.",
                "type": "string",
                "default": "gocity"
            },
            {
                "name": "ContainerPort",
                "description": "The number of port to expose on the container's IP address.",
                "type": "int",
                "default": 4000
            },
            {
                "name": "ServicePort",
                "description": "The port that will be exposed by this service.",
                "type": "int",
                "default": 4000
            }
        ],
        "stacks": [
            {
                "name": "dev",
                "fields": [
                    {
                        "name": "Image",
                        "description": "Docker image address of gocity. Default to 'yuanhao1223/gocity:latest'.",
                        "type": "string",
                        "default": "howieyuen/gocity:latest"
                    }
                ]
            }
        ]
    }
]

File: pkg/cmd/init/init.go [hunk 1]
```diff
@@ -27,6 +27,19 @@ var (
 
 		# Initialize a new KCL project from local directory
 		kusion init /path/to/templates`
+
+	templatesShort = `Templates used to initialize a new project`
```
Comment (SparkYuan) [inline pkg/cmd/init/init.go:31]: It's a little confusing about the current expression, maybe you could add a verb like show or list at the beginning of this sentence. e.g. "List all templates used to initialize a new project"

Comment (amyXia1994) [reply pkg/cmd/init/init.go]: +1, and the subcommand `kusion init templates` is also a little confusing, a subcommand like`kusion init list-templates` sounds more clear

File: pkg/cmd/init/init.go [hunk 2]
```diff
@@ -61,5 +74,36 @@ func NewCmdInit() *cobra.Command {
...
+
+	templatesCmd := newCmdTemplates()
+	cmd.AddCommand(templatesCmd)
+	return cmd
+}
+
+func newCmdTemplates() *cobra.Command {
+	o := NewTemplatesOptions()
+	cmd := &cobra.Command{
+		Use:                   "templates",
```
Comment (amyXia1994) [inline pkg/cmd/init/init.go:86]: the subcommand `kusion init templates` is also a little confusing, a subcommand like `kusion init list-templates` sounds more clear

**Comment by amyXia1994** [comment]
The error message should output to **stderr**, but now the wrong git repo url will result in an error ouput in **stdout**. :
`kusion init templates https://github.com/KusionStack/no-exist --online=true -o json`
 ❌  no templates

`kusion init templates KusionStack/kusion-templates.git --online=true -o json`
 ❌  template 'kusionstack/kusion-templates.git' not found

`kusion init templates https://github.com/kusion-templates.git --online=true -o json`
 ❌  failed to retrieve git folder: invalid Git URL

File: pkg/cmd/init/init.go (hunk 2 above)
Comment (howieyuen) [reply pkg/cmd/init/init.go:86]: how about `kusion templates`, strip `template` cmd out of `init`

**Review by amyXia1994 — APPROVED** [review]

**Pull request merged**